gateway-js/src/index.ts

```diff
@@ -580,20 +581,21 @@ export class ApolloGateway implements GraphQLService {
 
     this.compositionId = result.id;
     this.supergraphSdl = supergraphSdl;
+    this.supergraphSchema = supergraphSchema.toGraphQLJSSchema();
 
     if (!supergraphSdl) {
       this.logger.error(
         "A valid schema couldn't be composed. Falling back to previous schema.",
       );
     } else {
-      this.updateWithSchemaAndNotify(schema, supergraphSdl);
+      this.updateWithSchemaAndNotify(supergraphSchema, supergraphSdl);
 
       if (this.experimental_didUpdateSupergraph) {
         this.experimental_didUpdateSupergraph(
           {
             compositionId: result.id,
             supergraphSdl,
-            schema: schema.toGraphQLJSSchema(),
+            schema: this.schema!,
```

Comment:
**question** Should this be `this.supergraphSchema`? Previously it was "the thing returned by `createSchemaFromSupergraphSdl` with `toGraphQLJSSchema` called", and now it's gonna be "the thing returned by `createSchemaFromSupergraphSdl`"...right?